### ISSUE TITLE
Adding tagged attributes with integer values  does not follow the radius standard.

### ISSUE THREAD

**Issue by TobiasVi** [layeh/radius#42]
Hi,  
while analyzing some radius packets, generated with this radius library, I noticed that some length values are not correct. I compared the packet attributes with packets, generated by one other library and also with the radius standard.

How it should work, according to the standard:
When having a radius attribute that consist of a tag and an integer value, the tag is represented in one of the four bytes, that usually represent the integer value. That means, this integer value (in combination with tags) is allowed to use only 3 bytes.

What actually happens:
The tag gets its own byte, making the length of the attribute 1 byte larger than it is supposed to be.

This becomes evident when looking for example at the Tunnel-Medium-Type attribute in RFC 2868 (https://tools.ietf.org/html/rfc2868).  The tag has the size of 1 byte and the value a size of 3 byte. The total length is specified as 6. When adding this attribute to a radius packet using this layeh/radius library, the value of the length field is 7. This is because the value takes 4 bytes.

When looking at the implementation, the radius.NewTag() method seems to be the method that needs to be adjusted.

Greetings
Tobias

**Comment by sneakywombat** [comment]
I agree, I encountered this as well. The dot imports and layeh.com import paths should also be cleaned up to match the actual url this repo is hosted at.

**Comment by bontibon** [comment]
Thanks for the report, @TobiasVi.  I will try to address this as soon as I can.

> @sneakywombat 
> the dot imports

They are only used to make the code somewhat more readable in the examples; you do not have to do this in your own code.

If you discover an issue with them in the generated code, though, please open a ticket.

>  layeh.com import paths should also be cleaned up to match the actual url this repo is hosted at

Sorry, but that won't be changing.

**Comment by sneakywombat** [comment]
People tend to copy examples and bad examples mean more bad code :)  Just a suggestion, its a nit for sure.  One of the the better arguments for when to use and not to use dot imports: https://github.com/golang/go/wiki/CodeReviewComments#import-dot

As for the layeh.com, I know, that's why I forked it :)  It really is a limitation I needed to work around in my build system.  Don't misunderstand me though, you have a really great project here, thanks for your work!
